src/app/parcels/ParcelsPage.tsx

```diff
@@ -322,9 +322,14 @@ const ParcelsPage: React.FC<{}> = () => {
         lastStatus: rowToLastStatusColumn,
     };
 
-    const onParcelTableRowClick = (row: Row<ParcelsTableRow>): void => {
-        setSelectedParcelId(row.data.parcelId);
-        router.push(`/parcels?${parcelIdParam}=${row.data.parcelId}`);
+    const onParcelTableRowClick = (row: Row<ParcelsTableRow>, event: React.MouseEvent): void => {
+        const target = event.target as HTMLDivElement;
```

Comment:
I think it's quite hard to ensure this is indeed HTMLDivElement (in fact i'm not sure now). Let's keep it general for now
```suggestion
        const target = event.target as HTMLElement;
```